include/seqan3/io/stream/parse_condition.hpp

```diff
@@ -0,0 +1,1435 @@
+ * \brief An internal concept to check if an object fulfills the requirements of a seqan3::detail::parse_condition.
+ * \ingroup stream
+ *
+ * \details
+ *
+ * The must be invocable with an seqan3::char_adaptation_concept type and supply a static constexpr `msg` member of type
+ * seqan3::constexpr_string.
+ */
+//!\cond
+template <typename condition_t>
+concept bool parse_condition_concept = requires
+{
+    requires predicate_concept<std::remove_reference_t<condition_t>, char>;
+    requires std::is_base_of_v<parse_condition<remove_cvref_t<condition_t>>,
+                               remove_cvref_t<condition_t>>;
+
+    std::remove_reference_t<condition_t>::msg;
+
+    //The msg type can be added with a constexpr_string.
+    { constexpr_string<0>{} + std::remove_reference_t<condition_t>::msg } ->
+        decltype(std::remove_reference_t<condition_t>::msg);
+};
+//!\endcond
+
+/*!\name Requirements for seqan3::detail::parse_condition_concept
+ * \relates seqan3::detail::parse_condition_concept
+ * \brief You can expect the variable and the predicate function on all types that satisfy seqan3::ostream_concept.
+ * \{
+ */
+/*!\fn      bool operator()(char_type c);
+ * \brief   Predicate function to test if `c` satisfies the given condition.
+ * \ingroup stream
+ * \param   c The character to be tested.
+ * \returns `true` on success, `false` otherwise.
+ *
+ * \attention This is a concept requirement, not an actual function (however types satisfying this concept
+ * will provide an implementation).
+ */
+
+/*!\var static constexpr auto msg
+ * \memberof seqan3::detail::parse_condition_concept
+ * \brief Defines the condition msg. The type is deduced from the constant expression in the definition of the variable.
+ */
+//!\}
+
+// ----------------------------------------------------------------------------
+// make_printable
+// ----------------------------------------------------------------------------
+
+/*!\brief Returns a printable value for the given character `c`.
+ * \param[in] c The character to be represented as printable string.
+ * \return    a std::string containing a printable version of the given character `c`.
+ *
+ * \details
+ *
+ * Some characters, e.g. control commands cannot be printed. This function converts them to a std::string
+ * containing the visual representation of this character. For all control commands the value `'CTRL'` is returned.
+ *
+ * ### Exception
+ *
+ * Strong exception guarantee is given.
+ *
+ * ### Complexity
+ *
+ * Constant.
+ *
+ * ### Concurrency
+ *
+ * Thread-safe.
+ */
+template <typename char_type>
+inline std::string
+make_printable(char_type const c)
+{
+    if (c == '\0')
+        return "'\0'";
+    else if (c == '\t')
+        return "'\t'";
+    else if (c == '\n')
+        return "'\n'";
+    else if (c == '\v')
+        return "'\v'";
+    else if (c == '\f')
+        return "'\f'";
+    else if (c == '\r')
+        return "'\r'";
```

Comment:
You need double-`\` in all of these for the the "\" to bi visible OR you need to use raw string literals.